src/utils/skipRule.ts

```diff
@@ -176,11 +195,678 @@ function isSkipRulePassing(segment: SponsorTime | VideoLabelsCacheData, rule: Ad
     }
 }
 
+function isSkipPredicatePassing(segment: SponsorTime | VideoLabelsCacheData, predicate: AdvancedSkipPredicate): boolean {
+    if (predicate.kind === "check") {
+        return isSkipCheckPassing(segment, predicate as AdvancedSkipCheck);
+    } else { // predicate.kind === "operator"
+        // TODO Is recursion fine to use here?
+        if (predicate.operator == PredicateOperator.And) {
+            return isSkipPredicatePassing(segment, predicate.left) && isSkipPredicatePassing(segment, predicate.right);
+        } else { // predicate.operator === PredicateOperator.Or
+            return isSkipPredicatePassing(segment, predicate.left) || isSkipPredicatePassing(segment, predicate.right);
+        }
+    }
+}
+
 export function getCategoryDefaultSelection(category: string): CategorySelection {
     for (const selection of Config.config.categorySelections) {
         if (selection.name === category) {
             return selection;
         }
     }
     return { name: category, option: CategorySkipOption.Disabled} as CategorySelection;
-}
\ No newline at end of file
+}
+
+type TokenType =
+    | "if" // Keywords
+    | "disabled" | "show overlay" | "manual skip" | "auto skip" // Skip option
+    | `${SkipRuleAttribute}` // Segment attributes
+    | `${SkipRuleOperator}` // Segment attribute operators
+    | "and" | "or" // Expression operators
+    | "(" | ")" | "comment" // Syntax
+    | "string" | "number" // Literal values
+    | "eof" | "error"; // Sentinel and special tokens
+
```

Comment:
What are these tokens?

Reply:
`eof` is end of file (or end of input string, in this case), `error` is for any unexpected characters in the input

Reply:
The lexer (or tokenizer, it's a synonym) converts the input string (list of characters) into a list of tokens (basically words) in `nextToken`, which are then used by the parser in `parseConfig` (actually, it converts it one token at a time, because that's how the parser uses it)